runtime/doc/lua.txt

```diff
@@ -294,78 +294,6 @@ arguments separated by " " (space) instead of "\t" (tab).
         :luafile script.lua
         :luafile %
 <
-
-==============================================================================
-luaeval()                                                 *lua-eval* *luaeval()*
```

Comment:
To be honest, I'd probably leave this section here as-is (with only the `*lua-eval*` tag), as it's much more detailed than the "reference" style in `builtin.txt`. Instead, I'd add a brief function(!) documentation (arguments, return value etc.) with the `*luaeval()*` tag and add cross-links.

Reply:
I don't mind either way.

The reason I moved it out is because `lua.txt` is mostly useful for Lua usage, not vimscript. `luaeval()` is only useful from vimscript so made sense to bundle it with all the vimscript functions.

Reply:
That's a good point; it just seemed a bit "chatty" compared to the rest of the function documentation (and I don't think spending time on rewriting it to be more in line is worth it).

Reply:
I did think about that, but there are quite a few chatty descriptions anyway, e.g. `searchpair()`, `searchcount()`, `setqflist()`,

Reply:
Well, the static sections are valuable -- where would they go, and would they be able to stand alone?

Reply:
In either case, probably best treated as a follow-up (split `lua.txt` in generated and static sections).